experimental/packages/proto/src/metrics/types.ts

```diff
@@ -0,0 +1,281 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { IInstrumentationLibrary, IKeyValue, IResource } from '../common/types';
+
+/** Properties of an ExportMetricsServiceRequest. */
+export interface IExportMetricsServiceRequest {
+
+    /** ExportMetricsServiceRequest resourceMetrics */
+    resourceMetrics: IResourceMetrics[]
+}
+
+
+/** Properties of a ResourceMetrics. */
+export interface IResourceMetrics {
+    /** ResourceMetrics resource */
+    resource?: IResource;
+
+    /** ResourceMetrics instrumentationLibraryMetrics */
+    instrumentationLibraryMetrics: IInstrumentationLibraryMetrics[]
+
+    /** ResourceMetrics schemaUrl */
+    schemaUrl?: string;
+}
+
+
+/** Properties of an InstrumentationLibraryMetrics. */
+export interface IInstrumentationLibraryMetrics {
+
+    /** InstrumentationLibraryMetrics instrumentationLibrary */
+    instrumentationLibrary?: IInstrumentationLibrary;
+
+    /** InstrumentationLibraryMetrics metrics */
+    metrics: IMetric[];
+
+    /** InstrumentationLibraryMetrics schemaUrl */
+    schemaUrl?: string;
+}
+
+/** Properties of a Metric. */
+export interface IMetric {
+
+    /** Metric name */
+    name: string;
+
+    /** Metric description */
+    description?: string;
+
+    /** Metric unit */
+    unit?: string;
+
+    /** Metric gauge */
+    gauge?: IGauge;
+
+    /** Metric sum */
+    sum?: ISum;
+
+    /** Metric histogram */
+    histogram?: IHistogram;
+
+    /** Metric exponentialHistogram */
+    exponentialHistogram?: IExponentialHistogram;
+
+    /** Metric summary */
+    summary?: ISummary;
+}
+
+/** Properties of a Gauge. */
+export interface IGauge {
+    /** Gauge dataPoints */
+    dataPoints: INumberDataPoint[]
+}
+
+/** Properties of a Sum. */
+export interface ISum {
+
+    /** Sum dataPoints */
+    dataPoints: INumberDataPoint[];
```

Comment:
Maybe a linter auto fix is needed.

Reply:
i've run the auto fix but i'm not sure what issue you're trying to point to that should have been fixed?